std/file.d

```diff
@@ -1463,6 +1519,68 @@ version(Windows) @system unittest
     }
 }
 
+version(Posix) @system unittest
+{
+    import std.stdio : writefln;
+
+    auto currTime = Clock.currTime();
+
+    write(deleteme, "a");
+    scope(exit) assert(deleteme.exists), deleteme.remove;
+
+    SysTime accessTime1 = void;
+    SysTime modificationTime1 = void;
+    SysTime changeTime1 = void;
+
+    getTimesPosix(deleteme, accessTime1, modificationTime1, changeTime1);
+
+    enum leeway = 5.seconds;
+
+    {
+        auto diffa = accessTime1 - currTime;
+        auto diffm = modificationTime1 - currTime;
+        auto diffc = changeTime1 - currTime;
+        scope(failure) writefln("[%s] [%s] [%s] [%s] [%s] [%s] [%s]",
+                                accessTime1, modificationTime1, changeTime1, currTime, diffa, diffm, diffc);
+
+        assert(abs(diffa) <= leeway);
+        assert(abs(diffm) <= leeway);
+        assert(abs(diffc) <= leeway);
+    }
+
+    version(fullFileTests)
+    {
+        import core.thread;
+        enum sleepTime = dur!"seconds"(2);
+        Thread.sleep(sleepTime);
+
+        currTime = Clock.currTime();
+        write(deleteme, "b");
+
+        SysTime accessTime2 = void;
+        SysTime modificationTime2 = void;
+        SysTime changeTime2 = void;
+
+        getTimesPosix(deleteme, accessTime2, modificationTime2, changeTime2);
+
+        {
+            auto diffa = accessTime2 - currTime;
+            auto diffm = modificationTime2 - currTime;
+            auto diffc = changeTime2 - currTime;
+            scope(failure) writefln("[%s] [%s] [%s] [%s] [%s] [%s] [%s]",
+                                    accessTime2, modificationTime2, changeTime2, currTime, diffa, diffm, diffc);
+
+            //There is no guarantee that the access time will be updated.
+            assert(abs(diffa) <= leeway + sleepTime);
+            assert(abs(diffm) <= leeway);
+            assert(abs(diffc) <= leeway);
+        }
+
+        assert(accessTime1 <= accessTime2);
+        assert(modificationTime1 <= modificationTime2);
```

Comment:
Why `<=` and not `<` ? AFAIK the lowest precision of filesystem timestamps is FAT32 with 2 seconds, which should still almost always pass this test with `<`.

Reply:
Again, it's just copy-paste from getTimes. Not very good, but getTimesWin also has this copy-paste